genomepy/annotation.py

```diff
@@ -0,0 +1,395 @@
+    def genome_contigs(self):
+        if not self.__genome_contigs:
+            return get_column(self.sizes_file)
+
+    @property
+    def annotation_contigs(self):
+        return self.__annotation_contigs
+
+    @annotation_contigs.setter
+    def annotation_contigs(self, annotation_contig_list):
+        self.__annotation_contigs = annotation_contig_list
+
+    @annotation_contigs.getter
+    def annotation_contigs(self):
+        if not self.__annotation_contigs:
+            return list(set(get_column(self.annotation_bed_file)))
+
+    def filter_regex(
+        self,
+        annotation_file: str,
+        regex: Optional[str] = ".*",
+        invert_match: Optional[bool] = False,
+    ):
+        """
+        Filter annotation file (works with both gtf and bed) by contig name.
+
+        annotation_file: path to annotation file.
+        regex: regex string to keep.
+        invert_match: keep contigs NOT matching the regex string.
+
+        return a list of discarded contigs.
+        """
+        missing_contigs = []
+        tmp_dir = mkdtemp(dir=self.genome_dir)
+        filtered_annotation_file = os.path.join(
+            tmp_dir, os.path.basename(annotation_file)
+        )
+        with _open(annotation_file, "r") as old, _open(
+            filtered_annotation_file, "w"
+        ) as new:
+            for line in old:
+                if bool(re.match(regex, line)) is not invert_match:
+                    new.write(line)
+                else:
+                    missing_contigs.append(line.split("\t")[0])
+
+        shutil.move(filtered_annotation_file, annotation_file)
+        rm_rf(tmp_dir)
+        return list(set(missing_contigs))
+
+    def filter_genome_contigs(self):
+        """
+        filter the gtf for contigs in the genome.
+        return a list of discarded contigs.
+        """
+        missing_contigs = []
+        genome_contigs = self.genome_contigs
+        tmp_dir = mkdtemp(dir=self.genome_dir)
+        filtered_gtf_file = os.path.join(
+            tmp_dir, os.path.basename(self.annotation_gtf_file)
+        )
+        with _open(self.annotation_gtf_file, "r") as old, _open(
+            filtered_gtf_file, "w"
+        ) as new:
+            for line in old:
+                if not line.startswith("#"):
+                    gtf_contig_name = line.split("\t")[0]
+                    if gtf_contig_name in genome_contigs:
+                        new.write(line)
+                    else:
+                        missing_contigs.append(gtf_contig_name)
+
+        shutil.move(filtered_gtf_file, self.annotation_gtf_file)
+        rm_rf(tmp_dir)
+        return list(set(missing_contigs))
+
+    def _is_conforming(self):
+        """
+        Check if genome and annotation contigs conform.
+        Returns bool.
+        """
+        genome_contigs = self.genome_contigs
+        annotation_contigs = self.annotation_contigs
+        conforming = bool(set(genome_contigs) & set(annotation_contigs))
```

Comment:
You can use the self.attrs directly here

Reply:
```
def _is_conforming(self):
    return bool(set(self.genome_contigs) & set(self.annotation_contigs))
```

Reply:
What does it even do? Check if there are any overlapping contigs between the genome and annotaion?

Reply:
yes!